src/driver/odoc_driver.ml

```diff
@@ -704,50 +284,51 @@ module Dune_mode = struct
     Cmd.v info Term.(const run $ dune_style $ Common_args.term)
 end
 
-module Opam_mode = struct
-  let run packages packages_dir = run (Opam { packages; packages_dir })
+module OpamLibs = struct
+  let run libs = run (OpamLibs { libs })
+
+  let libs =
+    (* TODO: Is it package or library? *)
+    let doc = "The libraries to document" in
+    Arg.(value & opt_all string [] & info [ "l" ] ~doc)
+
+  let cmd_term = Term.(const run $ libs $ Common_args.term)
+
+  let cmd =
+    let doc =
+      "Documents libraries present in your opam switch. This mode will select  \n\
+      \               all libraries that are dependent on the selected \
+       libraries. This is a\n\
+      \               narrower set of dependencies than those chosen by \
+       invoking 'opam' mode."
+    in
+    let info = Cmd.info "opam-lib" ~doc in
+    Cmd.v info cmd_term
+end
+
+module OpamPackages = struct
+  let run packages = run (OpamPackages { packages })
 
   let packages =
     (* TODO: Is it package or library? *)
     let doc = "The packages to document" in
     Arg.(value & opt_all string [] & info [ "p" ] ~doc)
 
-  let packages_dir =
-    let doc = "Packages directory under which packages should be output." in
-    Arg.(
-      value
-      & opt (some Common_args.fpath_arg) None
-      & info [ "packages-dir" ] ~doc)
+  let cmd_term = Term.(const run $ packages $ Common_args.term)
 
   let cmd =
-    let doc = "Opam mode" in
+    let doc =
+      "Documents packages present in your opam switch. This mode will select \
+       all opam packages that are dependencies of the selected packages."
+    in
     let info = Cmd.info "opam" ~doc in
-    Cmd.v info Term.(const run $ packages $ packages_dir $ Common_args.term)
+    Cmd.v info cmd_term
 end
 
 let cmd =
   let doc = "Generate odoc documentation" in
   let info = Cmd.info "odoc_driver" ~doc in
-  Cmd.group info [ Voodoo_mode.cmd; Dune_mode.cmd; Opam_mode.cmd ]
-
-(* let map = Ocamlfind.package_to_dir_map () in
-   let _dirs = List.map (fun lib -> List.assoc lib map) deps in
-
-
-   let (_, lib_to_pkg_map) = Opam.pkg_to_dir_map () in
-   Opam.StringMap.iter (fun k v ->
-     if k <> v.Opam.name then
-       Format.printf "%s -> %a\n" k Opam.pp v) lib_to_pkg_map;
-   List.iter (fun dep -> Format.printf "%s\n%!" dep) deps;
-   ignore (exit 0); *)
-(* let all_units = all_units () in
-   let compiled = compile_all all_units in
-   let linked = link_all compiled in
-   let () = index_generate () in
-   (* let _ = js_index () in *)
-   ignore js_index;
-   let _ = Odoc.count_occurrences (Fpath.v "occurrences-from-odoc.odoc") in
-   ignore (generate_all linked);
-   let _ = Stats.bench_results () in *)
+  Cmd.group info
+    [ Voodoo_mode.cmd; Dune_mode.cmd; OpamPackages.cmd; OpamLibs.cmd ]
```

Comment:
By the way `Cmd.group` allows a default command: https://github.com/dbuenzli/cmdliner/blob/master/src/cmdliner.mli#L667 (you mentioned you wanted a "default" mode for the driver)

Reply:
It does, but it's not a great user experience - all of the options for the default command are listed, as well as the list of each of the subcommands, and it's a bit confusing.